src/octoprint/util/comm.py

```diff
@@ -3240,10 +3274,55 @@ def _send_loop(self):
 					# are done processing the last fetched queue entry
 					self._send_queue.task_done()
 
-				# now we just wait for the next clear and then start again
-				self._clear_to_send.wait()
+					if ( self._AdvancedOkSupported ):
+						if ( self._state == self.STATE_PRINTING 
```

Comment:
What about `STATE_PAUSING`, `STATE_RESUMING`, `STATE_CANCELLING`? Printing state is more than just `STATE_PRINTING`.

Reply:
this code filling the send queue with respect to marlin buffers, in non printing state older synchronous mode should be working fine

Reply:
The aforementioned states are also printing states however.